### ISSUE TITLE
Can hmmlearn (HMMGMM) be used for supervised learning?

### ISSUE THREAD

**Issue by AWin9** [hmmlearn/hmmlearn#109]
I am interested in HMM/GMM. I would like to use it for supervised learning, i.e. I have a sequence and labels for the sequence. Seqlearn allows us to use just MultinomialHMM. How can I implement hmmgmm for supervised learning using hmmlearn and seqlearn?
I want to solve this task http://cslu.ohsu.edu/~bedricks/courses/cs655/hw/hw4/hw4.html

**Comment by superbobry** [comment]
Hi, no `hmmlearn` does not currently target supervised learning. Have a look at [`seqlearn`](https://github.com/larsmans/seqlearn) instead. It does not implement a `GMMHMM`, but I think you get around it by combining `sklearn.mixture.GMM` with `sequlearn.hmm.MultinomialHMM`.

I've skimmed through the task you've linked and noticed some minor inaccuracies regarding the `hmmlearn` API. Specifically `hmmlearn` does not have an implementation of GMM, it re-uses the one from sckit-learn.

**Comment by AWin9** [comment]
Can the following method be successful?
1. We divide our set to subsets. The subset contains just samples of one class. For instance, we get 5 subsets.
2. We train particular model for every subset using EM (using fit(x)). We get 5 models.
3. For recognition, we evaluate likelihood for a sequence using every model separately (using score()). And the most value of the likelihood of some model will correspond some class.

**Comment by luigivieira** [comment]
Hi there @superbobry.

I found this discussion because I would like to use a GMMHMM in a supervised manner. I understand that I can not use seqlearn because my observations are not discrete. Your suggestion of using sklearn.mixture.GMM seems very interesting, but I don't even know how to start: I am new to HMM and GMM, and the documentation of scikit's GMM seems to indicate that the GMM is fit pretty much in the same way that it is done with hmmlearn: in a non-supervised fashion (i.e. without providing the labels for the samples).

Any help you could offer would be very appreciated. :)

**Comment by RoshanPAN** [comment]
@luigivieira have you get any idea of doing that?  I was doing similar thing (continuous observation as well), using GMM of sklearn and HMM (and multivariate  Gaussian distribution for modeling observation) in pomogranate. (But this gmm-hmm model does not work well for my case.) 
Did you find any other possible alternative solution to do this?

**Comment by luigivieira** [comment]
@RoshanPAN Nope, I didn't find any alternative solution to do this. :(

**Comment by chananshgong** [comment]
Hi @AWin9 , I did something similar by feeding the model pretrained gmm per state the transition matrix and the start probabilities. It would look something like:

            model = GMMHMM(n_mix=self.n_mix,
                           n_components=n_states,
                           init_params='',
                           params='')
            # starting probabilities
            model.startprob_ = start_prob
            # transition matrix
            model.transmat_ = prob_trans_mat
            # gmm parameters
            model.gmms_ = gmm_per_state
